hypergan.py

```diff
@@ -168,22 +173,60 @@
 
 hc.set("batch_size", args.batch)
 
+def sample_grid(sample_file, sess, config):
+    generator = get_tensor("g")[0]
+    y_t = get_tensor("y")
+    z_t = get_tensor("z")
+    dropout_t = get_tensor("dropout")
+
+
+    x = np.linspace(0,1, 4)
+    y = np.linspace(0,1, 6)
+
+    #z = np.mgrid[-3:3:0.75, -3:3:0.38*3].reshape(2,-1).T
+    #z = np.mgrid[-3:3:0.6*3, -3:3:0.38*3].reshape(2,-1).T
+    #z = np.mgrid[-6:6:0.6*6, -6:6:0.38*6].reshape(2,-1).T
+
+    z = np.mgrid[-1:1:0.6, -1:1:0.38].reshape(2,-1).T
+    #z = np.square(1/z) * np.sign(z)
+    #z = np.mgrid[0:1000:300, 0:1000:190].reshape(2,-1).T
+    #z = np.mgrid[-0:1:0.3, 0:1:0.19].reshape(2,-1).T
+    #z = np.mgrid[0.25:-0.25:-0.15, 0.25:-0.25:-0.095].reshape(2,-1).T
+    #z = np.mgrid[-0.125:0.125:0.075, -0.125:0.125:0.095/2].reshape(2,-1).T
+    #z = np.zeros(z_t.get_shape())
+    #z.fill(0.2)
+
+
+    sample = sess.run(generator, feed_dict={z_t: z, dropout_t: 1.0})
+    print(np.shape(sample))
+    #plot(self.config, sample, sample_file)
+    stacks = [np.hstack(sample[x*6:x*6+6]) for x in range(4)]
+    plot(config, np.vstack(stacks), sample_file)
+
+
+batch_no = 0
 
 def epoch(sess, config):
     batch_size = config["batch_size"]
     n_samples =  config['examples_per_epoch']
     total_batch = int(n_samples / batch_size)
+    global batch_no
     for i in range(total_batch):
+        sample_file="samples/grid-%06d.png" % (batch_no * total_batch + i)
+        #sample_grid(sample_file, sess, config)
```

Comment:
This is awesome but I commented it out by default.  Maybe it should go into a cli flag